x/blackbox/blackbox_test.py

```diff
@@ -0,0 +1,610 @@
+    prop_assert(lsig_res, lsig_res.stack_top(), x ** 2)
+    prop_assert(bad_lsig_res, bad_lsig_res.stack_top(), Encoder.hex0x(x ** 2))
+
+    prop_assert(app_res, app_res.max_stack_height(), 2)
+    prop_assert(app_log_res, app_log_res.max_stack_height(), 2)
+    prop_assert(lsig_res, lsig_res.max_stack_height(), 2)
+    prop_assert(bad_lsig_res, bad_lsig_res.max_stack_height(), 2)
+
+    prop_assert(app_res, app_res.status(), "PASS")
+    prop_assert(app_log_res, app_log_res.status(), "PASS")
+    prop_assert(lsig_res, lsig_res.status(), "PASS")
+    prop_assert(bad_lsig_res, bad_lsig_res.status(), "REJECT")
+
+    prop_assert(app_res, app_res.passed(), True)
+    prop_assert(app_log_res, app_log_res.passed(), True)
+    prop_assert(lsig_res, lsig_res.passed(), True)
+    prop_assert(bad_lsig_res, bad_lsig_res.passed(), False)
+
+    prop_assert(app_res, app_res.rejected(), False)
+    prop_assert(app_log_res, app_log_res.rejected(), False)
+    prop_assert(lsig_res, lsig_res.rejected(), False)
+    prop_assert(bad_lsig_res, bad_lsig_res.rejected(), True)
+
+    prop_assert(app_res, app_res.rejected(), False)
+    prop_assert(app_log_res, app_log_res.rejected(), False)
+    prop_assert(lsig_res, lsig_res.rejected(), False)
+    prop_assert(bad_lsig_res, bad_lsig_res.rejected(), True)
+
+    prop_assert(app_res, app_res.error(), False)
+    prop_assert(app_log_res, app_log_res.error(), False)
+    prop_assert(lsig_res, lsig_res.error(), False)
+    prop_assert(bad_lsig_res, bad_lsig_res.error(), True)
+    assert (
+        bad_lsig_res.error(
+            pattern="logic 0 failed at line 7: log not allowed in current mode"
+        )
+        is True
+    )
+    prop_assert(
+        bad_lsig_res, bad_lsig_res.error(pattern="log not allowed"), True
+    )
+    prop_assert(
+        bad_lsig_res, bad_lsig_res.error(pattern="WRONG PATTERN"), False
+    )
+
+    prop_assert(app_res, app_res.noError(), True)
+    prop_assert(app_log_res, app_log_res.noError(), True)
+    prop_assert(lsig_res, lsig_res.noError(), True)
+    assert (
+        "logic 0 failed at line 7: log not allowed in current mode"
+        in bad_lsig_res.noError()
+    )
+
+
+APP_SCENARIOS = {
+    "app_exp": {
+        "inputs": [()],
+        # since only a single input, just assert a constant in each case
+        "assertions": {
+            DRProp.cost: 11,
+            # int assertions on log outputs need encoding to varuint-hex:
+            DRProp.lastLog: Encoder.hex(2 ** 10),
+            # dicts have a special meaning as assertions. So in the case of "finalScratch"
+            # which is supposed to _ALSO_ output a dict, we need to use a lambda as a work-around
+            DRProp.finalScratch: lambda _: {0: 2 ** 10},
+            DRProp.stackTop: 2 ** 10,
+            DRProp.maxStackHeight: 2,
+            DRProp.status: "PASS",
+            DRProp.passed: True,
+            DRProp.rejected: False,
+            DRProp.noError: True,
```

Comment:
Why would you assert on all of `status`, `passed`, `rejected`, and `noError`? It seems needlessly repetitive

Reply:
Yes, it's repetitive. But I'm propagating the existing assertions in `DryRunTestMixin` up here and demonstrating their usage so anyone can pick and choose which ones they really need/want to use.